server/src-lib/Hasura/App.hs

```diff
@@ -211,6 +213,9 @@ runHGEServer ServeOptions{..} InitCtx{..} initTime = do
 
   authMode <- either (printErrExit . T.unpack) return authModeRes
 
+  schemaCache <- onNothing _icRebuildableSchemaCache
+                 (printErrExit "expected schema cache to be initialised")
```

Comment:
I would probably just pass the schema cache as a separate argument to this function rather than include it in the `InitCtx` wrapped in a `Maybe`. That way the type system can prevent this error from ever happening.

Reply:
The schema cache is now part of `InitCtx{..}` and is of type `Maybe (RebuildableSchemaCache Run)`. It is is a `Maybe` because we get `InitCtx{..}` from `initialiseCtx` which is used in operations which do not require the schema cache to be initialised like `HCExport`, `HCClean`.

We can take schema cache outside of `InitCtx{..}` ofcourse but I felt it is better to be here.

Thoughts?